payroll/views/hr_payslip_views.xml

```diff
@@ -109,7 +109,7 @@
                         string="Cancel Payslip"
                         name="action_payslip_cancel"
                         type="object"
-                        states="draft,confirm,verify"
+                        states="draft,confirm,verify,done"
```

Comment:
What you can probably do here is
1. create a computed field, allow_payslip_cancel, in hr.payslip that mirrors the value of the "cancel payslip" setting
2. leave this button as is and create another Cancel button which is only shown in Done state add an attribute to the effect of: 'invisible': [('allow_payslip_cancel', '=', False)]

Reply:
Will do this next commit. Thanks for the idea.